paasta_tools/cli/schemas/kubernetes_schema.json

```diff
@@ -432,6 +432,101 @@
                         }
                     }
                 },
+                "node_selectors_preferred": {
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "properties": {
+                            "weight": {
+                                "type": "integer"
+                            },
+                            "preferences": {
+                                "type": "object",
+                                "additionalProperties": false,
+                                "patternProperties": {
+                                    "^[a-zA-Z0-9]+[a-zA-Z0-9-_./]*[a-zA-Z0-9]+$": {
```

Comment:
just double-checking: this is the k8s label format regex, right?

if so, does jsonschema have a way for us to enforce how long this property could be (since labels can only be so many characters long)?

Reply:
hm. Probably? Label lengths are complicated though, since I think there's an overall length limit as well as a limit on the sections between the dots. I can probably find or come up with a regex for this.

Reply:
not too big of a deal if it's difficult to do now, we can always come back to this and figure out how to represent this in jsonchema (or express this in code in `paasta validate`)

Reply:
I guess I could put an overall length limit of 253+1+63 on this, but I'm not sure how valuable that is compared to a more complicated regex.

I copied this from https://github.com/Yelp/paasta/pull/3755/files#diff-3d2bc1f8f532cc633bdeb9a38648967bf24a429729d63ebe60c1551072d07dfaR355 so if we fix this, we'll want to fix this in both places.